### DISCUSSION TITLE
Complex form freezes,

### DISCUSSION THREAD

**Discussion by lukuser** [Budibase/budibase#9114] [Help]
Hello, I'm in the process of creating a rather large form with several steps. All worked well till recently. I now try to go back to steps previously created and tested, and I encounter some errors as well as very slow performance of the platform with some "Unresponsive" errors, out of memory errors, and "Failed to send request" error flood. I'd hate to loose the work already done on the design of this multi-step form, so any help would be greatly appreciated
![image](https://user-images.githubusercontent.com/81827963/208513481-d412638d-5c2c-4c10-9bdf-bb526e439887.png)
![image](https://user-images.githubusercontent.com/81827963/208513528-cb438607-ac27-4ece-9686-141565e52c2c.png)

**Comment by melohagan** [comment]
Hey @lukuser 

Sorry to hear you're having issues. There appears to be a related issue open here: https://github.com/Budibase/budibase/issues/9106

Would you be willing to share your app export to support@budibase.com ? Then I can take a look myself, although I'm heading off for Christmas soon!

**Reply by lukuser** [reply]
hello @melohagan,  thank you for your response. I sent the exported app to support.

**Reply by melohagan** [reply]
Thanks @lukuser As it's a SQL Server DB I'll not be able to access the data to truly see what's going on, however I can see there are a lot of nested data providers, and repeaters within one screen which is going to have performance implications.

It's possible that you are running into this issue: https://github.com/Budibase/budibase/issues/9143

Are you able to rollback to **v2.1.42** ?

**Comment by lukuser** [comment]
Hi @melohagan, I don't know how to revert all the way to 2.1.42. I found a revert button in app settings, but it only allows me to revert back to the previous version the app was in, before the last update (2.2.3). After that, the revert button is no longer there.

**Comment by dannyshin76** [comment]
Same issue here, my app works good in 2.1.42 but it spews out hundreds of that error when I test it on 2.2.9.

**Comment by lsbicyclic** [comment]
Same problem for me.
Even in version 2.1.42, but less often.

**Comment by lukuser** [comment]
Version 2.1.42 did the trick for me. @melohagan, please keep me posted on finding a solution for this - for now we'll stick with version 2.1.42.

**Comment by shogunpurple** [comment]
For any of these issues occurring, it would be good to get:
- Some screenshots of the failing requests
- An app export that allows us to reproduce the issue (internal DB preferred)
- To let us know which DB you are using, a little bit more about your app, any other relevant debug information.

**Reply by dannyshin76** [reply]
I did some investigation and I think I've found something. App export attached here https://github.com/Budibase/budibase/discussions/9159

**Comment by lukuser** [comment]
Hello @melohagan , is there an update on this issue?  We're developing a large app with Budibase and we're stuck on version 2.1.42 because of this issue. It would be nice to move onto the latest version and start using newer features. Thanks

**Reply by melohagan** [reply]
Hey @lukuser I believe there was a performance change since 2.2.15, with the latest being **2.2.19**.
Could you try that and see if it resolves your issues?

**Comment by lukuser** [comment]
Hey @melohagan , I tried the 2.2.19 on both on-premise and cloud version. The application works fine on 2.1.42, but once upgraded it's main screen freezes and then all other, smaller screens are also don't render, even though when the app started and I go to smaller screens firs, they work ok. This behaviour is different from the original issue, where controls rendered partially and then errors were popping up. Now, all I see is blank screen with the application header on top (see img) 
![image](https://user-images.githubusercontent.com/81827963/213781704-e9055dc1-cc46-4c78-adf3-4aa8406f293a.png)
I tried both chrome and firefox. All components seem to be there on the left, but clicking on any of them does not show anything in the design area. In preview mode, the same story. I see the first screen with a table to select a deal from, but once I select a deal, the /deal/:id screen is blank. When I revert back to 2.1.42, everything is working fine.

**Reply by lukuser** [reply]
Hey @melohagan , Update: the blank screen seems to be caused by an accordion plugin. I removed it and now the screen renders. Looks like it doesn't work when the app is upgraded. Now I'm back to the original issue of the series of "Failed to send request" errors -they happen only on one of the form steps in this screen (Step 2 of 7). Attached is the partial component structure of that step. It has 3 levels of data providers/repeaters among other things. 
![image](https://user-images.githubusercontent.com/81827963/213817053-c7a9666d-6852-47ac-991c-6a9a24dd737c.png)

**Reply by melohagan** [reply]
Hey @lukuser Thanks for this, could you also provide some screenshots of any errors in the Network/Console tab please? (Fn+F12 on Chrome)

You could also have a look at your different repeater blocks, and filters to make sure that they all still look valid and that none of the bindings have got into a weird state.

**Comment by lsbicyclic** [comment]
I also tried 2.2.22, and my complex forms still freeze.
Task manager shows google taking a lot of memory and processor.
Requests in network tab on the console keep working and do not end.

I'll keep version 2.1.42 for now, but we were about to use the app in production, and it is not reassuring ...

**Reply by shogunpurple** [reply]
Hi @lsbicyclic - are you using any custom plugins in your app?

**Reply by shogunpurple** [reply]
Also it would be good to get an app export of your app so that we can debug. Thank you

**Reply by lsbicyclic** [reply]
> Hi @lsbicyclic - are you using any custom plugins in your app?

Hello Martin,

not at all, but I have a lot of MySql queries on the form that freezes.

**Reply by lsbicyclic** [reply]
> Also it would be good to get an app export of your app so that we can debug. Thank you

Yes but i am using external databases, not budibase file system ...

**Reply by lsbicyclic** [reply]
Hello,
any update on this ?
Still did not find any workaround or solution.

**Comment by lukuser** [comment]
Hey @melohagan, I think I found a workaround for my issue. So on the step that was bugging, I had a hierarchy of repeater blocks with many to one, going 3 levels deep and that was repeated in parallel twice (so six repeater blocks) in each level since the parent was on the "many" side and I was on the :id screen of it, there was a possibility of only one child. but my repeaters were still set with the default row limit of 10. Since there was a lot of control of the lowest level of repeaters, it appears that the screen builder was pre-rendering empty control placeholders for each of the 10 rows  (x 10 parents x 10 of its parents,). I set the limit to 1 for each block, and the step rendered fine.

**Reply by lukuser** [reply]
So to summarize, with version 2.1.42, 3 levels deep repeater block hierarchy was working fine with the row limits at default 10, but with 2.2.19, this won't work.

**Reply by melohagan** [reply]
Ah that's interesting, thanks for investigating and giving the feedback! 

Are you using an external SQL database? If so, using custom SQL queries if possible will provide much better performance than nested repeater blocks.

**Reply by lukuser** [reply]
Yes, I'm using the SQL db. Thanks for the tip on the custom query  - I'll put that to use.
